### PR TITLE
TM-4980 Publishable Positions - FE Update for Future Integration

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
~Dual Merge: ~
~- [API PR](linkToAPIPR) - ⏱️~
~- [Mock PR](linkToMockPR) - ⏱️~

[TM-4980](https://metaphase.atlassian.net/browse/TM-4980)

- [x] Modify common Expand card to optionally not show some data points
that way we can hide the additional data calls behind that
- [x] Tiny spinner on expand portion when making additional calls - basically got a spinner working, but it was ugly. If they hate this I can implement it, but I think this loading animation looks much cleaner.
~- [ ] Read filters - Bid Cycle~
~- [ ] Write with filters - Bid Cycle~
~- [ ] Read additional card data~
~- [ ] Write additional card data~
- [x] Rename feature flag for PP parts not yet in prd

This PR is really just updating the UI for potential future api integrations

[TM-4980]: https://metaphase.atlassian.net/browse/TM-4980?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/PositionExpandableContent/PositionExpandableContent.jsx [hunk 1]
```diff
@@ -45,7 +48,8 @@ const PositionExpandableContent = ({ sections, form, tempHideEdit }) => {
     }
     const minScreenWidth = 1650;
     // Append additional fields to collapsed view to fill blank space on wider screens
-    if (minScreenWidth < windowWidth && sections.bodySecondary) {
+    if (appendAdditionalFieldsToBodyPrimary
```
Comment (elizabeth-jimenez) [inline src/Components/PositionExpandableContent/PositionExpandableContent.jsx:51]: long word, but hopefully descriptive

File: src/Components/PositionExpandableContent/PositionExpandableContent.jsx [hunk 2]
```diff
@@ -193,13 +211,19 @@ PositionExpandableContent.propTypes = {
       disableEdit: PropTypes.bool,
     }),
   }),
+  appendAdditionalFieldsToBodyPrimary: PropTypes.bool,
+  showLoadingSpinner: PropTypes.bool,
```
Comment (elizabeth-jimenez) [inline src/Components/PositionExpandableContent/PositionExpandableContent.jsx:215]: rename since not a spinner

File: public/config/config.json [hunk 3]
```diff
@@ -35,7 +35,6 @@
     "post_handshake": false,
     "projected_vacancy": true,
     "publishable_positions": true,
-    "publishable_positions_integration": true,
```
Comment (elizabeth-jimenez) [inline public/config/config.json:38]: desperately trying to reduce the number of FFs lol

File: src/sass/_spinner.scss [hunk 4]
```diff
@@ -107,6 +114,10 @@
   position: absolute;
   top: 40%;
 }
+.tm-spinner-standard-medium-center {
```
Comment (elizabeth-jimenez) [inline src/sass/_spinner.scss:117]: rename to remove standard?

File: src/Components/PublishablePositions/PublishablePositions.jsx [hunk 5]
```diff
@@ -29,6 +29,8 @@ const PublishablePositions = ({ viewType }) => {
 
   const dataHasErrored = useSelector(state => state.publishablePositionsHasErrored);
   const dataIsLoading = useSelector(state => state.publishablePositionsIsLoading);
+  const additionalDataIsLoading = false;
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositions/PublishablePositions.jsx:32]: set to true to see animation

Comment (elizabeth-jimenez) [reply src/Components/PublishablePositions/PublishablePositions.jsx:32]: Will eventually hook into the additional data calls, if needed

File: src/sass/_spinner.scss (hunk 4 above)
Comment (elizabeth-jimenez) [reply src/sass/_spinner.scss:117]: not even using anymore, so will remove whole thing

File: public/config/config.json (hunk 3 above)
Comment (scott062) [reply public/config/config.json:38]: Just to clarify that publishable_positions should be false in all upper envs once this is released?

File: src/Components/PositionExpandableContent/PositionExpandableContent.jsx [hunk 6]
```diff
@@ -75,100 +79,114 @@ const PositionExpandableContent = ({ sections, form, tempHideEdit }) => {
...
-              />
-            </Linkify>
-            <div className="word-count">
-              {/* eslint-disable-next-line react/prop-types */}
-              {sections.textarea.length} / 2,000
+      {
+        showLoadingAnimation ?
+          <div className="loading-animation--10">
+            <div className="loading-message pbl-20">
+            Pulling additional card data
```
Comment (scott062) [inline src/Components/PositionExpandableContent/PositionExpandableContent.jsx:92]: Rename message to something more user friendly - i think 'pulling' and 'card' sounds weird
"Loading additional data"
"Fetching data"
"Loading ..."
other... etc

**Comment by scott062** [comment]
Could we speed up the animation - makes me feel like the page is going to be slow with how the bar moves? Thinking if its faster it conveys faster loading even if the time it takes doesn't change

**Comment by scott062** [comment]
Double checked the other expandable component usages and still looks good - @elizabeth-jimenez Done with review, few comments

File: src/Components/PositionExpandableContent/PositionExpandableContent.jsx [hunk 7]
```diff
@@ -193,13 +211,19 @@ PositionExpandableContent.propTypes = {
...
+  appendAdditionalFieldsToBodyPrimary: PropTypes.bool,
+  showLoadingAnimation: PropTypes.bool,
   tempHideEdit: PropTypes.bool,
+  onShowMore: PropTypes.func,
 };
 
 PositionExpandableContent.defaultProps = {
   form: undefined,
   sections: undefined,
+  appendAdditionalFieldsToBodyPrimary: true,
```
Comment (scott062) [inline src/Components/PositionExpandableContent/PositionExpandableContent.jsx:223]: Default `false` ?

File: src/sass/_common.scss [hunk 8]
```diff
@@ -113,4 +119,21 @@ $values in $filters-sizing-config {
...
+  }
+}
+
+@keyframes loadingAnimation {
+  0% {
+    background-position: right;
+  }
+  100% {
+    background-position: left;
+  }
```
Comment (scott062) [inline src/sass/_common.scss:138]: 😎

File: src/Components/PositionExpandableContent/PositionExpandableContent.jsx (hunk 6 above)
Comment (elizabeth-jimenez) [reply src/Components/PositionExpandableContent/PositionExpandableContent.jsx:92]: Went with "Loading additional data"

File: src/Components/PositionExpandableContent/PositionExpandableContent.jsx (hunk 7 above)
Comment (elizabeth-jimenez) [reply src/Components/PositionExpandableContent/PositionExpandableContent.jsx:223]: `true` cause that is the legacy behavior

File: src/sass/_common.scss [hunk 9]
```diff
@@ -124,7 +124,7 @@ $values in $filters-sizing-config {
 @each $time in $animation-times-config {
   .loading-animation--#{$time} {
     background: linear-gradient(135deg, white 0% 30%, $primary-alt-light 30% 70%, white 70% 100%) right;
-    background-size: 400%;
+    background-size: 350%;
```
Comment (elizabeth-jimenez) [inline src/sass/_common.scss:127]: shortening this makes the animation start up faster after it "ends". Still looks good in AIM (was less noticeable over there cause faster and smaller)

**Review by scott062 — APPROVED** [review]

**Pull request merged**